pravega/pravega
# Issue 5122: Improving Windows Compatibility

I mistaken messed up the other branch, which ended up automatically closing the other PR (https://github.com/pravega/pravega)/pull/5123.

**Change log description**  

1. Added some documentation recognizing potentially slow unit tests (those involved in touching a large amount of files) and a possible fix.
2. Target the proper platform specific temp directory within `S3ProxyImpl`.
3. Surround a file clean procedure with a try/catch block to absorb the exception.

**Purpose of the change**  
Makes progress towards: #5122 

**What the code does**  

2. There are two benefits for this change. `S3ProxyImpl`  will save files relative to the root of the drive. Firstly for aesthetic reasons, we should avoid dumping files at the top level of someone's `C:` drive, and place them into the proper `Temp` folder. More importantly, because it is placed in a `tmp` folder at the root, Windows will place extra permissions on this folder, which can prevent the creation (and deletion) of files under it, causing the test to fail. 
3. This procedure often fails because Windows (I believe unlike Unix systems) automatically places a FileLock on files, which we later attempt to delete. 

It does not look like we do any extra work to try and recover from `IOException`s here, so it may be appropriate simply to log the exception instead of passing it along. One issue with this is that these temporary files may accumulate over time (Well, this happens anyways..).

**How to verify it**  
(Optional: steps to verify that the changes are effective)

Reference: https://github.com/pravega/pravega/pull/5123
